### PR TITLE
Explain how Z values are asssigned to mesh elements

### PR DESCRIPTION
**Pull request by DelazJ**
refs #6883, refs qgis/QGIS#45140
Completes #7729

### CODE REVIEW

File: docs/user_manual/working_with_mesh/mesh_properties.rst [hunk 1]
```diff
@@ -670,6 +670,85 @@ To interact with or edit a base mesh layer element, following tools are availabl
...
+ +                                       +-------------------------+-------------------------------+------------------------------------------+
+ |                                       | Yes                     | :guilabel:`Vertex Z value`    | Average of the selected vertices         |
+ +---------------------------------------+-------------------------+-------------------------------+------------------------------------------+
+ | Vertex on an unshared edge            | No                      | :guilabel:`Vertex Z value`    | Default or user defined                  |
+ +                                       +-------------------------+                               +------------------------------------------+
+ |                                       | Yes                     |                               | Average of the selected vertices         |
+ +---------------------------------------+-------------------------+-------------------------------+------------------------------------------+
+ | Vertex on an edge shared by two faces | ---                     | Mesh layer                    | Interpolated from the faces' vertices    |
```
Comment (vcloarec) [inline docs/user_manual/working_with_mesh/mesh_properties.rst:715]:
```suggestion
 | Vertex on an edge shared by two faces | ---                     | Mesh layer                    | Interpolated from the edge' vertices    |
```

**Review by vcloarec — CHANGES_REQUESTED** [review]

File: docs/user_manual/working_with_mesh/mesh_properties.rst (hunk 1 above)
Comment (DelazJ) [reply docs/user_manual/working_with_mesh/mesh_properties.rst:715]: Any reason to have removed the "vertex on unshared edge" rows?

File: docs/user_manual/working_with_mesh/mesh_properties.rst [hunk 2]
```diff
@@ -670,6 +670,85 @@ To interact with or edit a base mesh layer element, following tools are availabl
...
+ +---------------------------------------+-------------------------+-------------------------------+------------------------------------------+
+ | Vertex on a face                      | ---                     | Mesh layer                    | Interpolated from the face vertices      |
+ +---------------------------------------+-------------------------+-------------------------------+------------------------------------------+
+ | Vertex snapped to a 2D vector feature | ---                     | ---                           | No value                                 |
```
[the remark this answers is not in the document]
Comment (DelazJ) [reply docs/user_manual/working_with_mesh/mesh_properties.rst:719]: Should it be backported?

**Review by DelazJ — COMMENTED** [review]
Thanks @vcloarec 
Few questions below and also, is the table understandable or do you think of any better formatting?

File: docs/user_manual/working_with_mesh/mesh_properties.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (vcloarec) [reply docs/user_manual/working_with_mesh/mesh_properties.rst:719]: The fix should be backported (I just added the label).
If you ask for the doc, maybe it should be backported once the fix will be effectively backported?

File: docs/user_manual/working_with_mesh/mesh_properties.rst (hunk 1 above)
Comment (vcloarec) [reply docs/user_manual/working_with_mesh/mesh_properties.rst:715]: For vertex on edge inserted on an edge, the behavior is the same if the edge is shared or not by faces: the value is interpolated between the two vertices at the extremity of the edge.
So, better to remove also "shared by two faces" in line 711.

Comment (DelazJ) [reply docs/user_manual/working_with_mesh/mesh_properties.rst:715]: I'm not that sure. See video where numbers represent vertex z value and what happens (new vertex takes value from widget). Peek is bugging here so can't record but if I select vertices before adding the new one, I get the average value assigned.
https://user-images.githubusercontent.com/7983394/193492327-8ff3493a-4e27-4120-836c-59d555bcdde5.mp4

Comment (vcloarec) [reply docs/user_manual/working_with_mesh/mesh_properties.rst:715]: Indeed, may be a bug... I will look at...

File: docs/user_manual/working_with_mesh/mesh_properties.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (DelazJ) [reply docs/user_manual/working_with_mesh/mesh_properties.rst:719]: I was asking for code but your reply made me realize that I didn't backport any editing changes to 3.22 docs. So done now at #7818 (allowing you to review the unreviewed bits I merged last week - I'll forward port your suggestions)

**Pull request merged**